### PR TITLE
add _native_npu_attention support mask shape like [B,1,1,S]

### PR DESCRIPTION
**Pull request by chang-zhijie**
This PR resolves the unsupported atten_mask shape error when running attention with NPU (Ascend) devices.

Problem:
The NPU's fusion attention operator (e.g., npu_fusion_attention) does not support automatic broadcasting for attention masks.
When a mask of shape [batch, seq_len] or [batch, 1, 1, seq_len] is passed, the operator fails with an error similar to:
get unsupported atten_mask shape, the shape is [B, 1, 1, S] – while only shapes like [B, N, S, S], [B, 1, S, S], [1, 1, S, S], or [S, S] are accepted.

Solution:
When running on NPU, explicitly expand the mask to [B, 1, S, S] to satisfy the operator’s shape constraints.

Reference:
Ascend NPU fusion attention API:
https://www.hiascend.com/document/detail/zh/Pytorch/730/apiref/torchnpuCustomsapi/docs/context/torch_npu-npu_fusion_attention.md

Here is a  code for ErnieImage example of using the NPU backend:

```
python
import torch
import torch_npu
from diffusers import ErnieImagePipeline
from diffusers.utils import load_image

pipe = ErnieImagePipeline.from_pretrained("/model_dir/ERNIE-Image", torch_dtype=torch.bfloat16)
pipe = pipe.to("npu")
pipe.transformer.set_attention_backend("_native_npu")
generator = torch.Generator(device="npu")

prompt = "A black and white Chinese rural dog"
images = pipe(
    prompt=prompt,
    height=1024,
    width=1024,
    num_inference_steps=50,
    guidance_scale=5.0,
    generator=generator,
    use_pe=True,
).images
images[0].save("ernie-image-output.png")
```

### CODE REVIEW

File: src/diffusers/models/attention_dispatch.py [hunk 1]
```diff
@@ -1521,17 +1521,16 @@ def _maybe_modify_attn_mask_npu(query: torch.Tensor, key: torch.Tensor, attn_mas
...
-        and attn_mask.shape[0] == query.shape[0]
-        and attn_mask.shape[1] == key.shape[1]
-    ):
-        B, Sq, Skv = attn_mask.shape[0], query.shape[1], key.shape[1]
+    if attn_mask is not None:
+        if attn_mask.ndim == 2 and attn_mask.shape[0] == query.shape[0] and attn_mask.shape[1] == key.shape[1] :
+            batch_size, seq_len_q, seq_len_kv = attn_mask.shape[0], query.shape[1], key.shape[1]
+            attn_mask = attn_mask.unsqueeze(1).expand(batch_size, seq_len_q, seq_len_kv).unsqueeze(1).contiguous()
+        elif attn_mask.ndim == 4 and attn_mask.shape[1:3] == (1, 1):
+            attn_mask = attn_mask.expand(-1, -1, query.shape[1], -1)
```
Comment (yiyixuxu) [inline src/diffusers/models/attention_dispatch.py:1531]:
```suggestion
            attn_mask = attn_mask.expand(-1, -1, query.shape[1], -1).contiguous()
```

**Review by yiyixuxu — COMMENTED** [review]
i left one feedback, thanks!

**Comment by yiyixuxu** [comment]
@bot /style

File: src/diffusers/models/attention_dispatch.py (hunk 1 above)
Comment (chang-zhijie) [reply src/diffusers/models/attention_dispatch.py]: fix

**Review by yiyixuxu — APPROVED** [review]
thanks!

**Comment by yiyixuxu** [comment]
@bot /style

**Comment by yiyixuxu** [comment]
@chang-zhijie can you run `make style`? once CI passed I'll merge:)

**Comment by chang-zhijie** [comment]
@bot /style
